src/main/java/net/mcreator/ui/MCreatorApplication.java

```diff
@@ -263,6 +265,27 @@ public WorkspaceSelector getWorkspaceSelector() {
 	 */
 	public MCreator openWorkspaceInMCreator(File workspaceFile) {
 		this.workspaceSelector.setCursor(new Cursor(Cursor.WAIT_CURSOR));
+		String workspaceString = FileIO.readFileToString(workspaceFile);

```

Comment:
The code checking for the workspace lock and version should likely be moved to a Workspace.java in a static method